test/e2e/targets/awss3/main.go

```diff
@@ -0,0 +1,364 @@
+					var receivedObj []byte
+					var err error
+
+					By("polling the bucket objects", func() {
+						receivedObjs := e2es3.GetObjects(s3Client, bucketName)
+						Expect(receivedObjs).To(HaveLen(1),
+							"Received %d objects instead of 1", len(receivedObjs))
+
+						object := *receivedObjs[0]
+						receivedObj, err = ioutil.ReadAll(object.Body)
+						Expect(err).ToNot(HaveOccurred())
+					})
+
+					By("inspecting the object payload", func() {
+						ObjData := make(map[string]interface{})
+						err := json.Unmarshal(receivedObj, &ObjData)
+						Expect(err).ToNot(HaveOccurred())
+
+						eventData, err := json.Marshal(ObjData)
+						Expect(err).ToNot(HaveOccurred())
+
+						gotEvent := &cloudevents.Event{}
+						err = json.Unmarshal(eventData, gotEvent)
+						Expect(err).ToNot(HaveOccurred())
+
+						Expect(gotEvent.ID()).To(Equal(sentEvent.ID()))
+						Expect(gotEvent.Type()).To(Equal(sentEvent.Type()))
+						Expect(gotEvent.Source()).To(Equal(sentEvent.Source()))
+						Expect(gotEvent.Data()).To(Equal(sentEvent.Data()))
+						Expect(gotEvent.Extensions()[e2ece.E2ECeExtension]).
+							To(Equal(sentEvent.Extensions()[e2ece.E2ECeExtension]))
+					})
+				})
+			})
+		})
+
+		When("the CloudEvent context is discarded", func() {
+			BeforeEach(func() {
+				By("creating an AWSS3Target object with discardCEContext enabled", func() {
+					trgt, err := createTarget(trgtClient, ns, "test-",
+						withARN(bucketARN),
+						withCredentials(awsSecret.Name),
+						withDiscardCEContext(),
+					)
+					Expect(err).ToNot(HaveOccurred())
+
+					trgt = ducktypes.WaitUntilReady(f.DynamicClient, trgt)
+
+					trgtURL = ducktypes.Address(trgt)
+					Expect(trgtURL).ToNot(BeNil())
+				})
+			})
+			When("an event is sent to the target", func() {
+				var sentEvent *cloudevents.Event
+
+				BeforeEach(func() {
+					By("sending an event", func() {
+						sentEvent = e2ece.NewHelloEvent(f)
+
+						job := e2ece.RunEventSender(f.KubeClient, ns, trgtURL.String(), sentEvent)
+						apps.WaitForCompletion(f.KubeClient, job)
+					})
+				})
+
+				It("only puts the event's data onto the bucket object", func() {
+					var receivedObj []byte
+					var err error
+
+					By("polling the bucket objects", func() {
+						receivedObjs := e2es3.GetObjects(s3Client, bucketName)
+						Expect(receivedObjs).To(HaveLen(1),
+							"Received %d objects instead of 1", len(receivedObjs))
+
+						object := *receivedObjs[0]
+						receivedObj, err = ioutil.ReadAll(object.Body)
+						Expect(err).ToNot(HaveOccurred())
+					})
+					By("inspecting the object payload", func() {
+						Expect(receivedObj).To(Equal(sentEvent.Data()))
+					})
+				})
+			})
+		})
+	})
+
+	When("a client creates a target object with invalid specs", func() {
+
+		// Those tests do not require a real bucketARN or awsSecret
+		BeforeEach(func() {
+			bucketARN = "arn:aws:s3:eu-central-1:000000000000:test"
+			awsSecret = createAWSCredsSecret(f.KubeClient, ns, awsCreds)
```

Comment:
(related to my previous comment because it involves the `awsCreds` variable)

Creating a secret isn't required here since we're just testing the spec _validation_. We could just initialize `awsSecret` with an empty secret:

```golang
awsSecret = &corev1.Secret{}
```